### DISCUSSION TITLE
Any way to blur (with soft edges) Fabricjs Objects?

### DISCUSSION THREAD

**Discussion by kartiksolanki** [fabricjs/fabric.js#7773] [Q&A]
Let's say for example I have a triangle object as below which creates triangle with hard edges: 

var shadowTriangle = new fabric.Triangle({
      width: 150,
      height: 150,
      fill: 'black'
      originX: 'center',
      originY: 'center',
});

Is there any way/filter I can apply to make this triangle blurred with soft edges ??? Something like a light shadow beam ???

**Comment by melchiar** [comment]
Not natively no, masking support is on the roadmap but not implemented yet. 

You could always create a triangle subclass with a custom render function. The docs on subclassing are pretty sparse but you can take a look at [triangle.class.js](https://github.com/fabricjs/fabric.js/blob/master/src/shapes/triangle.class.js) to see how classes work.
